agent/engine/task_manager.go

```diff
@@ -423,6 +450,50 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 		mtask.Arn, container.Name, mtask.GetDesiredStatus().String())
 }
 
+// handleResourceStateChange attempts to update resource's known status depending on
+// the current status and errors during transition
+func (mtask *managedTask) handleResourceStateChange(resChange resourceStateChange) {
+	// locate the resource
+	res := resChange.resource
+	if !mtask.isResourceFound(res) {
```

Comment:
why do you need this check at all? The call path seems to indicate that this shouldn't be needed. If we need to do something like this, it's a major logic error in the agent and we should just `panic()` in that case. Either get rid of this or add enough tests to ensure that this doesn't ever happen.

Reply:
This is just another level of check to make sure things are sane. There is a similar check for container today here: https://github.com/aws/amazon-ecs-agent/blob/159ae5c3067e0a0d7387ff68af2a28bc90f3baaa/agent/engine/task_manager.go#L384 This scenario should not happen. I can remove the check if you have a strong preference.

Reply:
Let's log this at Critical level then (apologies if i asked you to change it to Error earlier.